index.ts

```diff
@@ -69,6 +69,18 @@ const RetryableHttpVerbs: string[] = ['OPTIONS', 'GET', 'DELETE', 'HEAD']
 const ExponentialBackoffCeiling = 10
 const ExponentialBackoffTimeSlice = 5
 
+export class HttpClientError extends Error {
+  constructor(message: string, statusCode: number) {
+    super(message)
+    this.name = 'HttpClientError'
+    this.statusCode = statusCode
+    Object.setPrototypeOf(this, HttpClientError.prototype)
```

Comment:
I took a short look into the pull request and was wondering what this line is about. Isn't it redundant? I thought the prototype of `this` (in a constructor) is `<class>.prototype` by default.

Reply:
Apparently it's to fix a TypeScript issue - https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work

Reply:
Interesting. Didn't know about this issue. Thanks for your reply.